Neos.Utility.Unicode/Classes/TYPO3/Flow/Utility/Unicode/Functions.php

```diff
@@ -179,6 +179,15 @@ public static function pathinfo($path, $options = null)
      */
     public static function parse_url($url, $component = -1)
     {
+        $hostFromUrl = parse_url($url, PHP_URL_HOST);
+        if ($hostFromUrl === false) {
+            return false;
+        }
+        $portFromUrl = parse_url($url, PHP_URL_PORT);
+        if ($portFromUrl === false) {
+            return false;
+        }
```

Comment:
we now call `parse_url()` three times. I guess for a good reason, but just to make sure: In which cases would the above checks return false and shouldn't there be a test covering these?

Reply:
Ok, removed one of the calls. As for when this may fail: just as in the call that was there before this change… if something goes wrong, handle it.